Reachable unwrap panic in `user_task_entry()`

### Describe the bug
There is a reachable unwrap panic in `user_task_entry()` at kernel/src/thread/task.rs:77.

https://github.com/asterinas/asterinas/blob/25a918d132277eff7ea202cacc4bfa284de2dec5/kernel/src/thread/task.rs#L77

This is because of the `EFAULT` in `write_val` at process/signal/mod.rs:166:

https://github.com/asterinas/asterinas/blob/25a918d132277eff7ea202cacc4bfa284de2dec5/kernel/src/process/signal/mod.rs#L166

### To Reproduce
Compile and run the PoC program:
```C
#define _GNU_SOURCE
#include <signal.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/mman.h>
#include <ucontext.h>
#include <unistd.h>

#define ALT_STACK_ADDR ((void *)0xdeadb000)
#define ALT_STACK_SIZE (0x1000)

void signal_handler(int signo, siginfo_t *info, void *context) { exit(0); }

int main() {
  stack_t ss;
  ss.ss_sp = ALT_STACK_ADDR;
  ss.ss_size = ALT_STACK_SIZE;
  ss.ss_flags = 0;

  if (mmap(ALT_STACK_ADDR, ALT_STACK_SIZE, PROT_NONE,
           MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0) == MAP_FAILED) {
    perror("mmap failed");
    return 1;
  }

  sigaltstack(&ss, NULL);
  perror("sigaltstack");

  // Set up the signal handler with SA_ONSTACK.
  struct sigaction sa;
  memset(&sa, 0, sizeof(sa));
  sa.sa_sigaction = signal_handler;
  sa.sa_flags = SA_ONSTACK | SA_SIGINFO;
  sigemptyset(&sa.sa_mask);

  sigaction(SIGSEGV, &sa, NULL);
  perror("sigaction");

  raise(SIGSEGV);

  return 0;
}
```

### Expected behavior
Asterinas shall handle this error and not panic.

### Environment
- Official docker asterinas/asterinas:0.9.4
- Intel(R) Xeon(R) Gold 6230R CPU @ 2.10GHz
- Asterinas version: main 25a918d

### Logs
```
[   193.024] ERROR: Uncaught panic: called `Result::unwrap()` on an `Err` value: Error { errno: EFAULT, msg: None }
at /root/asterinas/kernel/src/thread/task.rs:77
Printing stack trace:
   1: fn 0xffffffff8881c7b0 - pc 0xffffffff8881c81a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e0990;

   2: fn 0xffffffff8804a7f0 - pc 0xffffffff8804b50c / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e09f0;

   3: fn 0xffffffff8804a7e0 - pc 0xffffffff8804a7ea / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1010;

   4: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1020;

   5: fn 0xffffffff889fbcc0 - pc 0xffffffff889fbd04 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1030;

   6: fn 0xffffffff889f5a50 - pc 0xffffffff889f5b95 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1070;

   7: fn 0xffffffff88221f70 - pc 0xffffffff88222d57 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1170;

   8: fn 0xffffffff88560f90 - pc 0xffffffff88560f96 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1bf0;

   9: fn 0xffffffff886253f0 - pc 0xffffffff88625407 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1c00;

  10: fn 0xffffffff886280f0 - pc 0xffffffff88628108 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1c20;

  11: fn 0xffffffff88622390 - pc 0xffffffff886223bf / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1c40;

  12: fn 0xffffffff885e7210 - pc 0xffffffff885e7222 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1c90;

  13: fn 0xffffffff883ca670 - pc 0xffffffff883ca686 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1cc0;

  14: fn 0xffffffff88169f50 - pc 0xffffffff88169f5c / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1f00;

  15: fn 0xffffffff8855dee0 - pc 0xffffffff8855deee / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1f30;

  16: fn 0xffffffff888caf00 - pc 0xffffffff888caf28 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1f50;

  17: fn 0xffffffff88847f20 - pc 0xffffffff88847fe7 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e1f90;


     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi 0xffffffff889c6896; rdi 0xffffffff887ea837; rbp                0x0; rsp 0xffffdfffff9e2000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panic.rs:107
(  2) /root/asterinas/kernel/src/thread/oops.rs:116
(  3) ??:?
(  4) 4bdh3ctc2ehskuj9t0av352n0:?
(  5) ??:?
(  6) ??:?
(  7) /root/.rustup/toolchains/nightly-2024-10-12-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:1103
(  8) /root/.rustup/toolchains/nightly-2024-10-12-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
(  9) /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/unwinding-0.2.3/src/panicking.rs:56
( 10) egnwob4oghfgpic2tyf3fftva:?
( 11) /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/unwinding-0.2.3/src/panicking.rs:42
( 12) /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/unwinding-0.2.3/src/panic.rs:87
( 13) /root/asterinas/kernel/src/thread/oops.rs:54
( 14) /root/asterinas/kernel/src/thread/task.rs:95
( 15) /root/.rustup/toolchains/nightly-2024-10-12-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 16) /root/.rustup/toolchains/nightly-2024-10-12-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2454
( 17) /root/asterinas/ostd/src/task/mod.rs:162
make: *** [Makefile:194: run] Error 1
```

--- a/kernel/src/process/signal/mod.rs
+++ b/kernel/src/process/signal/mod.rs
@@ -17,6 +17,7 @@ use core::{mem, sync::atomic::Ordering};
 
 use align_ext::AlignExt;
 use c_types::{siginfo_t, ucontext_t};
+use constants::SIGKILL;
 pub use events::{SigEvents, SigEventsFilter};
 use ostd::{cpu::UserContext, user::UserContextApi};
 pub use pause::{with_signal_blocked, Pause};
@@ -46,7 +47,7 @@ pub fn handle_pending_signal(
     user_ctx: &mut UserContext,
     ctx: &Context,
     syscall_number: Option<usize>,
-) -> Result<()> {
+) {
     // We first deal with signal in current thread, then signal in current process.
     let posix_thread = ctx.posix_thread;
     let signal = {
@@ -54,7 +55,7 @@ pub fn handle_pending_signal(
         if let Some(signal) = posix_thread.dequeue_signal(&sig_mask) {
             signal
         } else {
-            return Ok(());
+            return;
         }
     };
 
@@ -93,8 +94,7 @@ pub fn handle_pending_signal(
             }
 
             drop(sig_dispositions);
-
-            handle_user_signal(
+            if let Err(e) = handle_user_signal(
                 ctx,
                 sig_num,
                 handler_addr,
@@ -103,7 +103,10 @@ pub fn handle_pending_signal(
                 mask,
                 user_ctx,
                 signal.to_info(),
-            )?
+            ) {
+                debug!("Failed to handle user signal: {:?}", e);
+                do_exit_group(TermStatus::Killed(SIGKILL));
+            }
         }
         SigAction::Dfl => {
             drop(sig_dispositions);
@@ -130,7 +133,6 @@ pub fn handle_pending_signal(
             }
         }
     }
-    Ok(())
 }
 
 #[allow(clippy::too_many_arguments)]

--- a/kernel/src/thread/task.rs
+++ b/kernel/src/thread/task.rs
@@ -87,12 +87,12 @@ pub fn create_new_user_task(
             if current_thread.is_exited() {
                 break;
             }
-            handle_pending_signal(user_ctx, &ctx, syscall_number).unwrap();
+            handle_pending_signal(user_ctx, &ctx, syscall_number);
             // If current is suspended, wait for a signal to wake up self
             while current_thread.is_stopped() {
                 Thread::yield_now();
                 debug!("{} is suspended.", current_posix_thread.tid());
-                handle_pending_signal(user_ctx, &ctx, None).unwrap();
+                handle_pending_signal(user_ctx, &ctx, None);
             }
             if current_thread.is_exited() {
                 debug!("exit due to signal");

